### DISCUSSION TITLE
Incompatibility/Unwanted Behaviour Notice: Plex Docker

### DISCUSSION THREAD

**Discussion by epd5** [tarampampam/error-pages#298] [General]
Running traefik:latest, plexinc/pms-docker & tarampampam/error-pages:3.2 on Docker.

Just wanted to make a note of a finding I had, any input very welcome.
I realize this isn't a bug or anything wrong with error-pages, just couldn't find anyone with this issue.

Problem:
My Plex server requires login even on the local network.
I was receiving a 401 Unauthorized notice when using error-pages and navigating to plex.mydomain.tld.

I don't have a great understanding of HTTP & how all the headers work, but I assume Plex 1st responds with a 401 Unauthorized when you 1st enter the plex containers web gui which then forwards you to the plex website login. Using error-pages, it must see this 401 response & take over immediately, giving you the 401 page & preventing access to Plex.

Workaround:
To stop this I had to set to` - traefik.http.middlewares.errorpages.errors.status=404-599`, from 400 - 599 to 400,402-599.
Another solution might be to invoke the ALLOWED_NETWORKS env variable in Plex, though I haven't tried this.

If there's a better workaround let me know.
All the best.

Edits: Corrections

**Comment by tarampampam** [comment]
Your diagnosis is spot on. Plex (like many apps with authentication flows) intentionally returns `401` as a signal to the client to redirect to the login page - it's not an "error" in the traditional sense, but part of a challenge/response handshake. Traefik's `errors` middleware can't distinguish between an intentional `401` and a real unauthorized error, so it intercepts it regardless.

Your workaround - excluding `401` from the status range - is exactly the right approach.

The same applies to any other app that uses a specific status code as part of its normal flow (e.g., some apps use `302` or `403` for redirects/access control).

No changes needed on the error-pages side - this is purely a Traefik middleware configuration decision about which status codes you want to intercept.
